### PR TITLE
Fix no-pie mode for compilers that default to PIE

### PR DESCRIPTION
**Pull request by loganmc10**
Fix for #34 

Tested on Ubuntu 17.04

### CODE REVIEW

**Comment by loganmc10** [comment]
We'll obviously the ```-no-pie``` option doesn't exist on older GCC versions, Travis failed. I'm not sure how to handle this situation

**Comment by loganmc10** [comment]
Fixed the Travis error by adding a check to see if PIE is the default for the compiler. Should be ok to merge now

File: projects/unix/Makefile [hunk 1]
```diff
@@ -178,6 +171,19 @@ MKDIR ?= mkdir -p
 COMPILE.c = $(Q_CC)$(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
 LINK.o = $(Q_LD)$(CC) $(CFLAGS) $(LDFLAGS) $(TARGET_ARCH)
 
+ifeq ($(PIE), 1)
+  CFLAGS += -fPIE
+  LDFLAGS += -pie
+else
+  CFLAGS += -fno-PIE
+  ifeq ($(CC),$(CROSS_COMPILE)gcc)
+    PIE_DEFAULT = $(shell gcc -v 2>&1 | grep enable-default-pie)
```
Comment (Narann) [inline projects/unix/Makefile]: Nice but I would add a tiny comment: "check if PIE is the default for the compiler"

Comment (loganmc10) [reply projects/unix/Makefile]: done:)

**Comment by richard42** [comment]
I think the new code block needs to go up in the makefile, because you're changing the CFLAGS/LDFLAGS variables after the place where they're used. Also, I think the LDFLAGS option should be "--no-pie", not "-no-pie"

**Comment by loganmc10** [comment]
They can go there because the other variables like LINK.o have been lazy set, so they don't use the value of LDFLAGS until they are expanded. If they were set using ```:=``` then it would need to go before. I put it there because I needed the value of CC.

I'll look into the no-pie option. I'll try --no-pie on my machine in a bit

**Comment by loganmc10** [comment]
I tried --no-pie, it didn't complain about that option, but it also didn't do anything. You can read here: https://wiki.ubuntu.com/SecurityTeam/PIE (including the gcc mailing list links), but I do believe that -no-pie is the right option.

**Comment by richard42** [comment]
Yeah, it could be that GCC 5.4.0 doesn't support -no-pie yet

**Pull request merged**